src/Momento.Sdk/Requests/Vector/VectorIndexItem.cs

```diff
@@ -0,0 +1,50 @@
+using System.Collections.Generic;
+using Momento.Sdk.Messages.Vector;
+
+namespace Momento.Sdk.Requests.Vector;
+
+/// <summary>
+/// A item in a vector index. Contains an ID, the vector, and any associated metadata.
+/// </summary>
+public class VectorIndexItem
```

Comment:
I don't think we can since we the metadata dictionary would be compared by reference. I tried making MetadataValue a record for a while but I couldn't overwrite the equals method in ListStringValue to make it compare correctly. Looks like I overlooked adding an equals and hashcode method here. I can add that in.